src/main/kotlin/com/zepben/evolve/streaming/grpc/Connect.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.streaming.grpc
+
+import com.zepben.auth.client.ZepbenTokenFetcher
+import com.zepben.auth.client.createTokenFetcher
+import com.zepben.auth.common.AuthMethod
+import java.io.File
+
+/**
+ * A collection of functions that return a channel that connects to Evolve's gRPC service, given address and authentication details.
+ */
+object Connect {
+
+    @JvmStatic
+    fun connectInsecure(
+        host: String = "localhost",
+        rpcPort: Int = 50051
+    ): GrpcChannel =
+        GrpcChannelBuilder().forAddress(host, rpcPort).build()
+
+    @JvmStatic
+    fun connectTls(
+        host: String = "localhost",
+        rpcPort: Int = 50051,
+        ca: File? = null
+    ): GrpcChannel =
+        GrpcChannelBuilder().forAddress(host, rpcPort).makeSecure(rootCertificates = ca).build()
+
+    @JvmStatic
+    fun connectWithSecret(
+        clientId: String,
+        clientSecret: String,
+        confAddress: String? = null,
+        confCAFilename: String? = null,
+        authCAFilename: String? = null,
+        host: String = "localhost",
+        rpcPort: Int = 50051,
+        ca: File? = null
+    ): GrpcChannel {
+        val tokenFetcher = createTokenFetcher(confAddress ?: "https://$host/ewb/auth", confCAFilename = confCAFilename, authCAFilename = authCAFilename)
+            ?: return connectTls(host, rpcPort, ca)
+
+        return connectWithSecretUsingTokenFetcher(tokenFetcher, clientId, clientSecret, host, rpcPort, ca)
+    }
+
+    @JvmStatic
+    fun connectWithSecret(
+        clientId: String,
+        clientSecret: String,
+        audience: String,
+        issuerDomain: String,
+        authMethod: AuthMethod,
```

Comment:
this can default to Auth0 (or OAuth???), as it doesn't make sense to provide an audience / issuerDomain without OAuth as an AuthMethod

Reply:
This creates a ambiguity if @JvmOverloads is active.

Reply:
Hm... how? audience and issuerDomain should still be required properties?

Reply:
Resolved after argument reordering. It used to be the case that `connectWithSecret(String, String, String, String)` in the JVM could reference either overload of the function.